types/html-to-text/index.d.ts

```diff
@@ -47,112 +118,291 @@ export interface HtmlToTextOptions {
+     *
+     * If numeric value is provided - it is used.
+     * Otherwise, if global `wordwrap` number is provided - it is used.
+     * If neither is true, then the fallback value of 40 is used.
+     */
+    length?: number;
+    /**
+     * (Only for: `blockquote` formatter.) Trim empty lines from blockquote.
+     */
+    trimEmptyLines?: boolean;
+    /**
+     * (Only for: `table`, `dataTable` formatter.) By default, heading cells (`<th>`) are uppercased.
+     *
+     * Set this to `false` to leave heading cells as they are.
+     */
+    uppercaseHeaderCells?: boolean;
+    /**
+     * (Only for: `table`, `dataTable` formatter.) Data table cell content will be wrapped to fit this width
+     * instead of global `wordwrap` limit.
+     *
+     * Set to `undefined` in order to fall back to `wordwrap` limit.
+     */
+    maxColumnWidth?: number;
+    /**
+     * (Only for: `table`, `dataTable` formatter.) Number of spaces between data table columns.
+     */
+    colSpacing?: number;
+    /**
+     * (Only for: `table`, `dataTable` formatter.) Number of empty lines between data table rows.
+     */
+    rowSpacing?: number;
 }
 
-export interface Formatters {
-    text?: LeafFormatter;
-    image?: LeafFormatter;
-    lineBreak?: Formatter;
-    paragraph?: Formatter;
-    anchor?: Formatter;
-    blockquote?: Formatter;
-    heading?: Formatter;
-    table?: Formatter;
-    orderedList?: Formatter;
-    unorderedList?: Formatter;
-    listItem?: Formatter;
-    horizontalLine?: Formatter;
+/**
+ * Simplified definition of [htmlparser2](https://github.com/fb55/htmlparser2) Node type.
+ *
+ * Makes no distinction between elements (tags) and data nodes (good enough for now).
+ */
+export interface DomNode {
+    /**
+     * Type of node - "text", "tag", "comment", "script", etc.
+     */
+    type: string;
+    /**
+     * Content of a data node.
+     */
+    data?: string;
+    /**
+     * Tag name.
+     */
+    name?: string;
+    /**
+     * Tag attributes dictionary.
+     */
+    attribs?: any;
+    /**
+     * Child nodes.
+     * Not optional for typescript use.
+     */
+    children: DomNode[];
+    /**
+     * Parent node.
+     */
+    parent?: DomNode;
 }
 
-export type LeafFormatter<T = any> = (
-    el: T,
-    options: HtmlToTextOptions
-) => string;
+/**
+ * A function to stringify a DOM node.
+ */
+export type FormatCallback = (elem: DomNode, walk: RecursiveCallback,
+    builder: BlockTextBuilder, formatOptions: FormatOptions) => void;
+
+/**
+ * A function to process child nodes.
+ * Passed into a {@link FormatCallback} as an argument.
+ */
+export type RecursiveCallback = (nodes: DomNode[], builder: BlockTextBuilder) => void;
 
-export type Formatter<T = any> = (
-    el: T,
-    walk: (dom: any[], options: HtmlToTextOptions) => string,
-    options: HtmlToTextOptions
-) => string;
+/**
+ * Type of object passed to tags in the options.
+ */
+export interface TagDefinitions {
+    // ''?: TagDefinition; How to support?
```

Comment:
```suggestion
    ''?: TagDefinition;
```
This seems fine for now.

Reply:
Retesting.  I thought I had a ts error leading to the question.

Reply:
String literal just like the others.... 🤷‍♂️

Reply:
Maybe a cache problem in vs code was showing a linting error.  Can't remember for sure now.  Restarting vs code sometimes clears the problems tab.